src/openvic-simulation/country/CountryInstance.cpp

```diff
@@ -176,20 +277,155 @@ bool CountryInstance::apply_history_to_country(CountryHistoryEntry const* entry)
 	for (Reform const* reform : entry->get_reforms()) {
 		ret &= add_reform(reform);
 	}
+	set_optional(tech_school, entry->get_tech_school());
+	constexpr auto set_bool_map_to_indexed_map =
+		[]<typename T>(IndexedMap<T, bool>& target, ordered_map<T const*, bool> source) {
+			for (auto const& [key, value] : source) {
+				target[*key] = value;
+			}
+		};
+	set_bool_map_to_indexed_map(technologies, entry->get_technologies());
+	set_bool_map_to_indexed_map(inventions, entry->get_inventions());
+	// entry->get_foreign_investment();
+
+	// These need to be applied to pops
+	// entry->get_consciousness();
+	// entry->get_nonstate_consciousness();
+	// entry->get_literacy();
+	// entry->get_nonstate_culture_literacy();
+
+	set_optional(releasable_vassal, entry->is_releasable_vassal());
+	// entry->get_colonial_points();
+	for (std::string const& flag : entry->get_country_flags()) {
+		ret &= set_country_flag(flag);
+	}
+	for (std::string const& flag : entry->get_global_flags()) {
+		// TODO - set global flag
+	}
+	// entry->get_government_flag_overrides();
+	for (Decision const* decision : entry->get_decisions()) {
+		// TODO - take decision
+	}
 
 	return ret;
 }
 
+void CountryInstance::_update_production() {
+
+}
+
+void CountryInstance::_update_budget() {
+
+}
+
+void CountryInstance::_update_technology() {
+
+}
+
+void CountryInstance::_update_politics() {
+
+}
+
+void CountryInstance::_update_population() {
+	total_population = 0;
+	national_literacy = 0;
+	national_consciousness = 0;
+	national_militancy = 0;
+	pop_type_distribution.clear();
+
+	for (auto const& state : states) {
+		total_population += state->get_total_population();
```

Comment:
I'd store `state->get_total_population()` in a variable `state_population` and reuse that instead of calling it 4 times.

Reply:
Sure, I'll do that both here and in `State.cpp` and I'll also change it from an `int32_t` conversion to a `fixed_point_t` conversion. Eventually I'm expecting we'll change `Pop::pop_size_t` to a fixed point type rather than `int64_t`, as I believe it'll be needed for keeping track of population growth (otherwise small pops would never grow).